python/src/load.cpp

```diff
@@ -361,3 +420,25 @@ void mlx_savez_helper(
 
   return;
 }
+
+void mlx_save_safetensor_helper(
+    py::object file,
+    py::dict d,
+    bool retain_graph) {
+  auto arrays_map = d.cast<std::unordered_map<std::string, array>>();
+  if (py::isinstance<py::str>(file)) {
+    save_safetensor(py::cast<std::string>(file), arrays_map, retain_graph);
+    return;
+  } else if (is_ostream_object(file)) {
+    auto writer = std::make_shared<PyFileWriter>(file);
+    {
+      py::gil_scoped_release gil;
+      save_safetensor(writer, arrays_map, retain_graph);
+    }
+
+    return;
+  }
+
+  throw std::invalid_argument(
+      "[save_safetensor] Input must be a file-like object, string, or pathlib.Path");
```

Comment:
😅  I had no idea we supported `pathlib.Path`, that's awesome 👍

Reply:
Probably good to add a test which also loads the temp file name in a `pathlib.Path` to have that case covered for the future.

Reply:
Originally copied this error message from `mlx_load_helper` but after testing we don't actually support `Path` so will correct the error messages. Can come back and add support for it tho in a later PR